commit 91d08b11a72764f6d199f32fa0c1b1abd4edc2ad
Author: PGZXB <pgzxb@qq.com>
Date:   Tue Feb 10 17:09:30 2026 +0000

    Fix heap buffer overlfow in opj_j2k_read_sod

diff --git a/src/lib/openjp2/j2k.c b/src/lib/openjp2/j2k.c
index a2014c89..ad76aca7 100644
--- a/src/lib/openjp2/j2k.c
+++ b/src/lib/openjp2/j2k.c
@@ -5082,12 +5082,14 @@ static OPJ_BOOL opj_j2k_read_sod(opj_j2k_t *p_j2k,

         OPJ_UINT32 l_current_tile_part =
             l_cstr_index->tile_index[p_j2k->m_current_tile_number].current_tpsno;
-        l_cstr_index->tile_index[p_j2k->m_current_tile_number].tp_index[l_current_tile_part].end_header
-            =
-                l_current_pos;
-        l_cstr_index->tile_index[p_j2k->m_current_tile_number].tp_index[l_current_tile_part].end_pos
-            =
-                l_current_pos + p_j2k->m_specific_param.m_decoder.m_sot_length + 2;
+        if (l_cstr_index->tile_index[p_j2k->m_current_tile_number].tp_index &&
+                l_current_tile_part <
+                l_cstr_index->tile_index[p_j2k->m_current_tile_number].nb_tps) {
+            l_cstr_index->tile_index[p_j2k->m_current_tile_number].tp_index[l_current_tile_part].end_header
+                = l_current_pos;
+            l_cstr_index->tile_index[p_j2k->m_current_tile_number].tp_index[l_current_tile_part].end_pos
+                = l_current_pos + p_j2k->m_specific_param.m_decoder.m_sot_length + 2;
+        }

         if (OPJ_FALSE == opj_j2k_add_tlmarker(p_j2k->m_current_tile_number,
                                               l_cstr_index,
