diff --git a/setup.py b/setup.py
index 1a31a89..b21f975 100644
--- a/setup.py
+++ b/setup.py
@@ -212,7 +212,7 @@ class pil_build_ext(build_ext):
             # if Homebrew is installed, use its lib and include directories
             import subprocess
             try:
-                prefix = subprocess.check_output(['brew', '--prefix']).strip()
+                prefix = subprocess.check_output(['brew', '--prefix']).strip().decode('latin1')
             except:
                 # Homebrew not installed
                 prefix = None
